tests/org.eclipse.ui.workbench.texteditor.tests/src/org/eclipse/ui/workbench/texteditor/tests/FindReplaceDialogTest.java

```diff
@@ -146,4 +147,87 @@ public void testShiftEnterReversesSearchDirectionDialogSpecific() {
 		dialog.simulateEnterInFindInputField(true);
 		assertEquals(5, (target.getSelection()).x);
 	}
+
+	@Test
+	public void testReplaceAndFindAfterInitializingFindWithSelectedString() {
+		openTextViewer("text text text");
+		getTextViewer().setSelectedRange(0, 4);
+		initializeFindReplaceUIForTextViewer();
+		DialogAccess dialog= getDialog();
+
+		assertThat(dialog.getFindText(), is("text"));
+
+		IFindReplaceTarget target= dialog.getTarget();
+		assertEquals(0, (target.getSelection()).x);
+		assertEquals(4, (target.getSelection()).y);
+
+		dialog.performReplaceAndFind();
+
+		assertEquals(" text text", getTextViewer().getDocument().get());
+		assertEquals(1, (target.getSelection()).x);
+		assertEquals(4, (target.getSelection()).y);
+	}
+
+	@Test
+	public void testIncrementalSearchOnlyEnabledWhenAllowed() {
+		initializeTextViewerWithFindReplaceUI("text text text");
+		DialogAccess dialog= getDialog();
+
+		dialog.select(SearchOptions.INCREMENTAL);
+		dialog.select(SearchOptions.REGEX);
+
+		dialog.assertSelected(SearchOptions.INCREMENTAL);
+		dialog.assertDisabled(SearchOptions.INCREMENTAL);
+	}
+
+	/*
+	 * Test for https://github.com/eclipse-platform/eclipse.platform.ui/pull/1805#pullrequestreview-1993772378
+	 */
+	@Test
+	public void testIncrementalSearchOptionRecoveredCorrectly() {
+		initializeTextViewerWithFindReplaceUI("text text text");
+		DialogAccess dialog= getDialog();
+
+		dialog.select(SearchOptions.INCREMENTAL);
+		dialog.assertSelected(SearchOptions.INCREMENTAL);
+		dialog.assertEnabled(SearchOptions.INCREMENTAL);
+
+		reopenFindReplaceUIForTextViewer();
+		dialog= getDialog();
+		dialog.assertSelected(SearchOptions.INCREMENTAL);
+		dialog.assertEnabled(SearchOptions.INCREMENTAL);
+
+		dialog.select(SearchOptions.REGEX);
+		dialog.assertSelected(SearchOptions.INCREMENTAL);
+		dialog.assertDisabled(SearchOptions.INCREMENTAL);
+
+		reopenFindReplaceUIForTextViewer();
+		dialog= getDialog();
+		dialog.assertSelected(SearchOptions.INCREMENTAL);
+		dialog.assertDisabled(SearchOptions.INCREMENTAL);
+	}
+
+	@Test
+	public void testFindWithWholeWordEnabledWithMultipleWordsNotIncremental() {
+		initializeTextViewerWithFindReplaceUI("two words two");
+		DialogAccess dialog = getDialog();
+		dialog.setFindText("two");
+		dialog.select(SearchOptions.WHOLE_WORD);
+		dialog.select(SearchOptions.WRAP);
+		IFindReplaceTarget target= dialog.getTarget();
+
+		dialog.simulateEnterInFindInputField(false);
+		assertEquals(0, (target.getSelection()).x);
+		assertEquals(3, (target.getSelection()).y);
```

Comment:
Is there any reason for changing the behavior of the test? This is the code of the original test
```suggestion
		assertEquals(0, (target.getSelection()).x);
		assertEquals(0, (target.getSelection()).y);
```

Reply:
@HeikoKlare The proposed change would (,... and the old test did) simply fail, and that makes sense!

We start in a non-incremental position, set a find-text, press enter and expect the first occurrence of the text to be selected.

The reason there was some back-and-forth with this test is that at some point, I tested that "nothing is selected before pressing enter", which is not relevant to testing the whole-word behavior of the Dialog.

I COULD add a new test which tests for "what happens to the selection when I open the Dialog and do nothing" but I doubt that that is relevant enough, unless we spot bugs in the future

Reply:
Thank you. The reason for the change seems sound to me, so it's acceptable to have it in this PR. I am just hestitant to combine a refactoring with a functional change, at least in case both are actually independent from each other (as in this case where one does not require the other).

> The proposed change would (,... and the old test did) simply fail, and that makes sense!

That's not correct, as the existing test obviously succeeds 🙂

Reply:
With proposed change I meant the proposed change to 

```java
		assertEquals(0, (target.getSelection()).x);
		assertEquals(0, (target.getSelection()).y);```

Reply:
Yes, that proposed change leads to the exact behavior of the existing test, which does obviously not fail. The PR introduced a slight modification of the test behavior, but as mentioned, it's fine.